### DISCUSSION TITLE
Selectively processing images in folder based on file size

### DISCUSSION THREAD

**Discussion by nickharambee** [ImageMagick/ImageMagick#5411] [Help] [answered]
Hello,

I am using the following command in terminal on a Mac to batch reduce the file size and dimensions of images:

cd [path] && mogrify -define jpeg:extent=500kb -resize 4096x2160\> *.jpg

I am wondering if it possible to amend this command so that only images larger than 500kb are processed (i.e. images smaller than 500kb would be skipped). Or amend it in such a way that the file size cannot be increased.

The reason I am asking is that sometimes the file size of small images (say 100kb) is actually increased by using the command, which I wan't to avoid.

Thanks,

Nick

**Comment by fmw42** [comment, marked as answer]
I do not know of a direct way with mogrify. You would have to write a script to filter out those files that you do not want to change before running mogrify.

Perhaps your issue with adding a profile or having large profiles.  Try adding -strip to your command

`mogrify -strip -define jpeg:extent=500kb -resize 4096x2160> *.jpg`

**Reply by nickharambee** [reply]
Thanks. Adding -strip meant that file sizes were not increased.
